test/lib/rules/no-unsupported-ssr-properties.js

```diff
@@ -107,6 +107,95 @@ testRule('no-unsupported-ssr-properties', {
                 }
             `,
         },
+        {
+            code: `
+                import { LightningElement } from 'lwc';
+                
+                export default class Foo extends LightningElement {
+                  connectedCallback() {
+                    if (!import.meta.env.SSR && !randomOtherCheck) {
+                      this.querySelector('span').getAttribute('role');
+                    }
+                  }
+                }
+            `,
+        },
+        {
+            code: `
+              import { LightningElement } from 'lwc';
+              
+              export default class Foo extends LightningElement {
+                connectedCallback() {
+                  if (!import.meta.env.SSR && randomOtherCheck) {
+                    this.querySelector('span').getAttribute('role');
+                  }
+                }
+              }
+          `,
+        },
+        {
+            code: `
+              import { LightningElement } from 'lwc';
+              
+              export default class Foo extends LightningElement {
+                connectedCallback() {
+                  if (randomOtherCheck && !import.meta.env.SSR) {
+                    this.querySelector('span').getAttribute('role');
+                  }
+                }
+              }
+          `,
+        },
+        {
+            code: `
+              import { LightningElement } from 'lwc';
+              
+              export default class Foo extends LightningElement {
+                connectedCallback() {
+                  if (!a && b && !c && d && !import.meta.env.SSR) {
+                    this.querySelector('span').getAttribute('role');
+                  }
+                }
+              }
+          `,
+        },
+        {
+            code: `
+              import { LightningElement } from 'lwc';
+              
+              export default class Foo extends LightningElement {
+                connectedCallback() {
+                  if (a && (b && !import.meta.env.SSR)) {
+                    this.querySelector('span').getAttribute('role');
+                  }
+                }
+              }
+          `,
+        },
+        {
+            code: `
+              import { LightningElement } from 'lwc';
+
+              export default class Foo extends LightningElement {
+                connectedCallback() {
+                  return isCSR ? this.template.querySelector('button') : null;
+                }
+              }
+          `,
+        },
```

Comment:
This doesn't look correct at all. This test is only passing because it's using `this.template.querySelector` rather than `this.querySelector`. We should open a separate issue to handle `this.template.querySelector` the same as `this.querySelector`.

```suggestion
        {
            code: `
              import { LightningElement } from 'lwc';

              export default class Foo extends LightningElement {
                connectedCallback() {
                  return isCSR ? this.template.querySelector('button') : null;
                }
              }
          `,
        },
```

Reply:
- https://github.com/salesforce/eslint-plugin-lwc/issues/174